slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c

```diff
@@ -58,11 +61,59 @@ void basictests() {
     return;
 #else
 	// faking it
-    firmware_check_done = 1;
+    //firmware_check_done = 1;
+	FILE_LOG(logINFOBLUE, ("******** Mythen3 Server: do the checks *****************\n"));
+	if (mapCSP0() == FAIL) {
+    	strcpy(firmware_message,
+				"Could not map to memory. Dangerous to continue.\n");
+		FILE_LOG(logERROR, ("%s\n\n", firmware_message));
+		firmware_compatibility = FAIL;
+		firmware_check_done = 1;
+		return;
+    }
+	// does check only if flag is 0 (by default), set by command line
+	if ((!debugflag) && ((testFpga() == FAIL))) {
+		strcpy(firmware_message,
+				"Could not pass basic tests of FPGA and bus. Dangerous to continue.\n");
+		FILE_LOG(logERROR, ("%s\n\n", firmware_message));
+		firmware_compatibility = FAIL;
+		firmware_check_done = 1;
+		return;
+	}
+
```

Comment:
I would include all the firmware and software checks as you already have an api version now.